File: Baekjoon/Math/[5893]17배/solution.js
const fileName = process.platform === "linux" ? "/dev/stdin" : "example.dat";
const inputData = require("fs")
  .readFileSync(fileName)
  .toString()
  .trim()
  .split("\n");

const mul17 = (n) => {
  let left = n + "0000";
  let right = "0000" + n;
  let result = "";

  let carry = false;
  for (let i = left.length - 1; i >= 0; i--) {
    if (left[i] == "0" && right[i] == "0") {
      if (carry) {
        result = "1" + result;
      } else {
        result = "0" + result;
      }
      carry = false;
    } else if (left[i] == "1" && right[i] == "1") {
      if (carry) {
        result = "1" + result;
      } else {
        result = "0" + result;
      }
      carry = true;
    } else {
      if (carry) {
        result = "0" + result;
        carry = true;
      } else {
        result = "1" + result;
        carry = false;
      }
    }
  }

  result = carry ? "1" + result : result;

  return result;
};

console.log(mul17(inputData));

File: Baekjoon/Math/[5893]17배/solution.py
import sys

N = sys.stdin.readline()


def mul17(n):
    left = n[:len(n) - 1] + "0000"
    right = "0000" + n
    result = ""

    carry = False
    for i in range(len(left) - 1, -1, -1):
        if left[i] == '0' and right[i] == '0':
            if carry:
                result = "1" + result
            else:
                result = "0" + result
            carry = False
        elif left[i] == '1' and right[i] == '1':
            if carry:
                result = "1" + result
            else:
                result = "0" + result
            carry = True
        else:
            if carry:
                result = "0" + result
                carry = True
            else:
                result = "1" + result
                carry = False

    result = "1" + result if carry else result

    return result


sys.stdout.write(mul17(N) + "\n")
